tt_metal/common/math.hpp

```diff
@@ -22,7 +22,7 @@ inline uint32_t round_up(uint32_t a, uint32_t b) { return b * div_up(a, b); }
 inline constexpr std::uint32_t round_down(std::uint32_t a, std::uint32_t b) { return a / b * b; }
 
 inline uint32_t positive_pow_of_2(uint32_t exponent) {
-    TT_FATAL(exponent >= 0 && exponent < 32);
+    TT_FATAL(exponent >= 0 && exponent < 32, "Exponent {} out of bounds.", exponent);
```

Comment:
@bbradelTT how commited are you to this PR?

Ideally message should include the valid range.

Reply:
The range is hard coded and will be shown in the message as the condition string.

I think it's important to fix the cases where the fatals were not doing anything.

There were enough failures in pipelines were assumptions were invalid that this should be addressed.

Other reviewers wanted to have TT_FATAL take two parameters to avoid the issue from happening again. As long as that can be done efficiently, I think it's a good idea.

However, spending time hand crafting over 1000 error messages, especially when the condition is printed, is not a good way to spend our time.